src/python/WMCore/MicroService/MSPileup/MSPileup.py

```diff
@@ -98,18 +99,36 @@ def updatePileup(self, pdict):
         """
         MSPileup update API to update corresponding pileup document in data layer.
 
-        :param pdict: input MSPilup data dictionary
+        :param pdict: input MSPileup data dictionary or partial pileup spec
```

Comment:
Alan, according to YOUR own comment (https://github.com/dmwm/WMCore/pull/11807#issuecomment-1860329756) it is either full document or a spec in a form of `{containerFraction: value, pileupName: string}`. The comment in docstring it using `or`, but I'll add `either` to make it more clear.

Reply:
My understanding is that pileup document update works has been implemented as a dict update. In other words, the user can provide a sub-set of key/value pairs, the server will then fetch the document from the database, update the document in memory, and persist the new document in the database.

If this is the way YOU (;)) have implemented it, then I think we can no longer guarantee whether the document (or dictionary) that reaches this updatePileup() method is complete or just a small set of key/value pairs. I think this needs to be made clear if we want to avoid future misunderstandings.

Reply:
No, the update works only on full document, we had discussion about it before and agree that users should provide full document which will be validated. Therefore, the provided document should satisfies the schema and as such it can be either full document (even though it may not have all optional fields) or partial spec. This is how it is implemented now.

Reply:
Then I think we should have stuck with that behavior and not allow partial update for the container fraction.
Unless I am missing something, the implementation seems to accept any sub-set of key/value pairs (when validate=true):
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/MicroService/MSPileup/MSPileupData.py#L151

Reply:
Alan, instead of arguing about this behavior is would be more productive if you'll clearly define how update should work in general case. If you want to make update for whole document, including this particular use-case, then you must define what is a mandatory key should be present in document passed via PUT method. Then, I can make appropriate changes. I think the confusion comes from the fact that validation just checks the dict key:value pairs, but it is not suppose to define update API behavior (in fact it is another way around, first we should define update API and then properly adjust validation).